packages/trpc/server/routers/loggedInViewer/outOfOffice.handler.test.ts

```diff
@@ -0,0 +1,1396 @@
+        reasonId: 2,
+        toUserId: users[0].id,
+        createdAt: new Date().toISOString(),
+        updatedAt: new Date().toISOString(),
+      },
+      {
+        userId: users[3].id,
+        uuid: uuidv4(),
+        start: getFutureDateUTC(20, TimeOfDay.START),
+        end: getFutureDateUTC(20, TimeOfDay.END),
+        notes: "",
+        reasonId: 1,
+        toUserId: users[1].id,
+        createdAt: new Date().toISOString(),
+        updatedAt: new Date().toISOString(),
+      },
+      {
+        userId: users[3].id,
+        uuid: uuidv4(),
+        start: getFutureDateUTC(22, TimeOfDay.START),
+        end: getFutureDateUTC(22, TimeOfDay.END),
+        notes: "",
+        reasonId: 2,
+        toUserId: users[0].id,
+        createdAt: new Date().toISOString(),
+        updatedAt: new Date().toISOString(),
+      },
+    ],
+  });
+}
+
+describe("outOfOfficeHandler", () => {
+  setupAndTeardown();
+
+  describe("outOfOfficeCreateOrUpdate", () => {
+    it("should successfully create outofoffice for logged-in user.", async () => {
+      const users = [
+        {
+          id: 1,
+          name: "User 1",
+          username: "user-1",
+          email: "user-1@test.com",
+          timeZone: "Asia/Kolkata",
+          schedules: [TestData.schedules.IstWorkHours],
+        },
+      ];
+      addUsers(users);
+
+      const { schedules, ...loggedInUser } = users[0];
+      const ctx = {
+        user: loggedInUser as NonNullable<TrpcSessionUser>,
+      };
+
+      const input: TOutOfOfficeInputSchema = {
+        dateRange: {
+          startDate: new Date("2024-10-01T00:00:00.000Z"),
+          endDate: new Date("2024-10-01T00:00:00.000Z"),
+        },
+        offset: 330,
+        toTeamUserId: null,
+        reasonId: 1,
+      };
+
+      const result = await outOfOfficeCreateOrUpdate({ ctx, input });
+      expect(result).toStrictEqual({});
+
+      //expect outOfOffice to be created in db for userId:1
+      const ooo = await prismock.outOfOfficeEntry.findMany({
+        where: {
+          userId: users[0].id,
+        },
+      });
+      expect(ooo.length).toEqual(1);
+
+      //expect the start and end dates are stored in db are correct
+      expect(ooo[0].start).toEqual("2024-10-01T00:00:00.000Z");
+      expect(ooo[0].end).toEqual("2024-10-01T23:59:59.999Z");
+    });
+
+    it("should successfully create outofoffice with redirect for logged-in user.", async () => {
+      const teams = [
+        {
+          id: 1,
+          name: "Team 1",
+          slug: "team-1",
+        },
+      ];
+      await addTeamsToDb(teams);
+
+      const users = [
+        {
+          id: 1,
+          name: "User 1",
+          username: "user-1",
+          email: "user-1@test.com",
+          timeZone: "Asia/Kolkata",
+          schedules: [TestData.schedules.IstWorkHours],
+        },
+        {
+          id: 2,
+          name: "User 2",
+          username: "user-2",
+          email: "user-2@test.com",
```

Comment:
We already have end-to-end tests for this common operation related to OOO entries. Please test the specific changes you made and focus primarily on team OOO cases. The current changes make the test unnecessarily larger. Also, use 1 or 2 items in the list or array of items as examples.

Reply:
Since we are modifying the same function to create ooo for individual user as well as for team admin, I think it's better to cover both scenarios for unit tests.

I have used minimum possible users for each scenario, for team admin creating ooo scenario - we need atleast three - 
1. admin (context user), 
2. user for whom ooo is being created
3. user for whom ooo is redirected.

In this test, I have used only 3.